init_process.go

```diff
@@ -0,0 +1,409 @@
+		// log.Fatalln("Cannot get bucket region")
+		fmt.Println("Cannot get bucket region")
+	}
+
+	bucket_region := aws.StringValue(output.LocationConstraint)
+	// Buckets in Region us-east-1 have a LocationConstraint of null.
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#GetBucketLocationOutput
+	if bucket_region == "" {
+		bucket_region = "us-east-1"
+	}
+
+	//create downloader
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(bucket_region)},
+	)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot creat a new session")
+	}
+	// need to specify session region!
+	// sess := session.Must(session.NewSession())
+	// fmt.Println(*sess.Config.Region)
+	s3_downloader := s3manager.NewDownloader(sess)
+
+	//download file and store
+	s3_file_name := strings.SplitN(s3_file_path, "/", -1)
+	file_from_s3, err := os.Create(ConfigFileFolderPath + "/" + s3_file_name[len(s3_file_name)-1])
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalln("Cannot creat s3 config file to store config info")
+	}
+	defer file_from_s3.Close()
+
+	_, err = s3_downloader.Download(file_from_s3,
+		&s3.GetObjectInput{
+			Bucket: aws.String(bucket_name),
+			Key:    aws.String(s3_file_path),
+		})
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot download %s from s3\n", s3_file_name)
+	}
+}
+
+//process config files user specified.
+func processAllConfigFiles(folderPath string) {
+	fileInfos, err := ioutil.ReadDir(folderPath)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Unable to read config files in %s folder\n", folderPath)
+	}
+
+	for _, file := range fileInfos {
+		content_b, err := ioutil.ReadFile(folderPath + "/" + file.Name())
+		if err != nil {
+			fmt.Println(err)
+			log.Fatalf("Cannot open file: %s\n", folderPath+"/"+file.Name())
+		}
+		content := string(content_b)
+		if strings.Contains(content, "[PARSER]") {
+			// this is a parser config file. change command
+			changeCommandToAddR("/" + folderPath + "/" + file.Name())
+		} else {
+			// this is not a parser config file. @INCLUDE
+			// writeInclude("/"+folderPath+"/"+file.Name(), MainConfigFilePath)
+			writeInclude("/"+folderPath+"/"+file.Name(), MainConfigFilePath)
+		}
+	}
+}
+
+//add @INCLUDE in main config file to include all customer specified config files
+func writeInclude(configFilePath, mainConfigFilePath string) {
+	mainConfigFile, err := os.OpenFile(mainConfigFilePath, os.O_APPEND|os.O_WRONLY, 0777)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Unable to read main config file: %s\n", mainConfigFilePath)
+	}
+	defer mainConfigFile.Close()
+
+	writeContent := "@INCLUDE " + configFilePath + "\n"
+	_, err = mainConfigFile.WriteString(writeContent)
+	if err != nil {
+		fmt.Println(err)
+		log.Fatalf("Cannot write %s in main config file: %s\n", writeContent[:len(writeContent)-2], mainConfigFilePath)
+	}
+}
+
+//change the cammand if needed.
+//add modified command to the init_process_entrypoint.sh
+func changeCommandToAddR(parserFilePath string) {
```

Comment:
maybe we can name it `updateCommand`

Reply:
Yea, will modify. And thank you for all the comments mentioned above :)